src/Api/FormatsProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Api;
+
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Anthony GRASSIOT <antograssiot@free.fr>
+ */
+final class FormatsProvider implements FormatsProviderInterface
+{
+    private $configuredFormats;
+    private $formats;
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, array $formats)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->configuredFormats = $formats;
+        $this->formats = [];
+    }
+
+    /**
+     * Finds formats from a Resource class.
+     */
+    public function getFormatsFromAttributes(array $attributes): array
+    {
+        if (!$attributes) {
+            return $this->configuredFormats;
+        }
+
+        $resourceMetadata = $this->resourceMetadataFactory->create($attributes['resource_class']);
+
+        if (!$formats = (array) $resourceMetadata->getOperationAttribute($attributes, 'formats', [], true)) {
```

Comment:
About `(array)`, is there a way to get something different from an array?

Reply:
Yes I guess by using `"formats"="json"` instead of `"formats"={"json"}`

Reply:
But as the key is plural, the user could easily understand it expects an array. Otherwise if the value is not an array, maybe throw an exception? WDYT @meyerbaptiste?

Reply:
I can if you all prefer, it was almost costless